Sources/ProcedureKitMobile/Alert.swift

```diff
@@ -76,53 +109,82 @@ public class AlertProcedure: UIProcedure {
      */
     @available (iOS 9.0, *)
     public var preferredAction: UIAlertAction? {
-        get { return alert.preferredAction }
-        set { alert.preferredAction = newValue }
+        get { return controller.preferredAction }
+        set { fatalError("Set the preferred action using the add(actionWithTitle:style:isPreferred:handler:) method.") }
     }
 
     /**
-     The array of text fields displayed by the alert. (read-only)
+     Creates an `AlertProcedure`. It must be constructed with the view
+     controller which the alert will be presented from. This is stored as a
+     weak reference.
 
-     Use this property to access the text fields displayed by the alert. The text fields are in the order in which you added them to the alert controller. This order also corresponds to the order in which they are displayed in the alert.
+     - parameter title: an optional alert title, defaults to nil
+     - parameter message: an optional alert message, defaults to nil
+     - parameter style: the `UIAlertControllerStyle`, defaults to .alert
+     - parameter from: a `UIViewController` instance.
+     - parameter waitForDismissal: a Bool, defaults to true, which indicates whether the
+          procedure should wait until the alert controller is dismissed until finishing.
 
-     - returns: the optional array of UITextField instances
+     - notes: The presenting view controller is weakly held.
      */
-    public var textFields: [UITextField]? {
-        return alert.textFields
+    public init(title: String? = nil, message: String? = nil, style: UIAlertControllerStyle = .alert, from viewController: PresentingViewController, waitForDismissal: Bool = true) {
```

Comment:
@danthorpe I think that we need to remove the option of changing the style from the Init method. Either that or extend the procedure to currently support the action sheet style on both iPhone and iPad. The current version will crash. Due to a missing presentation controller

Reply:
Yeah, is this because it needs to be put in a pop-over presentation controller?

Reply:
thats right, otherwise you'll get a crash on the iPad, you also nee do configure the presentation controller with the relevant data or it will still crash

Reply:
you need something along this lines;


let alertController = UIAlertController(title: "aaaaaa", message: "bbbbbbbb", preferredStyle: .actionSheet)
        alertController.addAction(UIAlertAction(title: "test", style: UIAlertActionStyle.destructive, handler: { (some) in
            
        }))
        present(alertController, animated: true, completion: nil)
        
        if let presentationPopover = alertController.popoverPresentationController {
            
            if let sourceView = ShareHelper.defaultsShareHelper.sourceView, !ShareHelper.defaultsShareHelper.sourceRect.isEmpty {
                
                presentationPopover.sourceView = sourceView
                presentationPopover.sourceRect = ShareHelper.defaultsShareHelper.sourceRect
                
            } else if let barButtonItem = ShareHelper.defaultsShareHelper.barButtonItem {
                
                presentationPopover.barButtonItem = barButtonItem
            }
        }
        present(alertController, animated: true) {
            ShareHelper.defaultsShareHelper.sourceRect = CGRect.zero
            ShareHelper.defaultsShareHelper.sourceView = nil
            ShareHelper.defaultsShareHelper.barButtonItem = nil
        }


Note that you can pass two different combinations to the popoverPresentationController. 
Think that it will be simpler to just make a different procedure for this, also did you ever try to use the actionSheet style in your code? it's not really a common use case

Reply:
Yeah, I agree - I think we'll fix `AlertProcedure` to just alerts.